frontend/src/components/Item/Metadata/ImageSearch.vue

```diff
@@ -90,179 +90,174 @@
   </v-dialog>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
 import {
   ImageProviderInfo,
   RemoteImageInfo,
   ImageType,
   BaseItemDto
 } from '@jellyfin/sdk/lib/generated-client';
 import { getRemoteImageApi } from '@jellyfin/sdk/lib/utils/api/remote-image-api';
-import { useResponsiveClasses } from '@/composables';
+import { useRemote, useResponsiveClasses } from '@/composables';
 
-export default defineComponent({
-  props: {
-    metadata: {
-      type: Object,
-      default: (): BaseItemDto => ({})
-    },
-    dialog: {
-      type: Boolean,
-      default: false
-    }
+const props = defineProps<{
+  metadata: BaseItemDto;
+  dialog: boolean;
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:dialog', isOpen: boolean): void;
+  (e: 'download-success', someting: boolean): void;
+}>();
+
+const { t } = useI18n();
+const remote = useRemote();
+
+const providers = ref<ImageProviderInfo[]>([]);
+const type = ref<ImageType>(ImageType.Primary);
+// eslint-disable-next-line unicorn/no-null -- the v-select component uses null to represent no value
+const source = ref<ImageProviderInfo['Name'] | null>(null);
+const allLanguages = ref(false);
+const images = ref<RemoteImageInfo[]>([]);
+const loading = ref(true);
+
+const types = computed((): { text: string; value: ImageType }[] => [
```

Comment:
I left this in because TypeScript will infer the type as union of the different values in this list (which are a set of unique strings). Which means that theoretically someone could add an entry to this list which has a value that isn't a valid `ImageType`. Including this type prevents that since there will an error if that happens. Keeping types as narrow as possible is good practice so I've added it in.